### PR TITLE
Add control-related JITServer changes

### PR DESCRIPTION
**Pull request by harryyu1994**
- Add JITServer/JITClient CHTable and IProfiler initializations
- Add JITServer's J9HOOK_VM_INITIALIZED routine
- Add TR_PrintJITServerMsgStats, TR_PrintJITServerCHTableStats and TR_PrintJITServerIPMsgStats routines for JITServer's remote messages, CHTable and IProfiler stats
- Add various JITServer persistent collection initializations (i.e. _unloadedClassesTempList, _newlyExtendedClasses)

Signed-off-by: Harry Yu <harryyu1994@gmail.com>

### CODE REVIEW

**Comment by harryyu1994** [comment]
`compInfo->incrementLocalGCCounter()`
https://github.com/eclipse/openj9/commit/25e7b9690287074a517099415c4491296de68f78#diff-88fcba8cc3e43212acb5330efde03ef0

File: runtime/compiler/control/rossa.cpp [hunk 1]
```diff
@@ -1820,11 +1863,27 @@ aboutToBootstrap(J9JavaVM * javaVM, J9JITConfig * jitConfig)
          {
          javaVM->sharedClassConfig->runtimeFlags &= ~J9SHR_RUNTIMEFLAG_ENABLE_AOT;
          TR_J9SharedCache::setSharedCacheDisabledReason(TR_J9SharedCache::AOT_DISABLED);
+#if defined(JITSERVER_SUPPORT)
+         if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
+            {
+            // TODO: Format the error message to use j9nls_printf
+            fprintf(stderr, "Aborting Compilation: SCC/AOT must be enabled and can be stored in JITServer Server Mode.");
```
Comment (mpirvu) [inline runtime/compiler/control/rossa.cpp]: The message is not clear at all here. We are not aborting a compilation, but the entire JVM. Maybe something like: "Error: -Xaot:nostore option is not compatible with JITServer mode" is better.

File: runtime/compiler/control/rossa.cpp [hunk 2]
```diff
@@ -1820,11 +1863,27 @@ aboutToBootstrap(J9JavaVM * javaVM, J9JITConfig * jitConfig)
...
          }
       else if ((javaVM->sharedClassConfig->runtimeFlags & J9SHR_RUNTIMEFLAG_ENABLE_AOT) == 0)
          {
          TR::Options::getAOTCmdLineOptions()->setOption(TR_NoStoreAOT);
          TR_J9SharedCache::setSharedCacheDisabledReason(TR_J9SharedCache::AOT_DISABLED);
+#if defined(JITSERVER_SUPPORT)
+         if (compInfo->getPersistentInfo()->getRemoteCompilationMode() == JITServer::SERVER)
+            {
+            // TODO: Format the error message to use j9nls_printf
+            fprintf(stderr, "Aborting Compilation: SCC/AOT must be enabled and can be stored in JITServer Server Mode.");
```
Comment (mpirvu) [inline runtime/compiler/control/rossa.cpp]: Same here: "Error: -Xnoaot option must not be specified for JITServer"

**Comment by mpirvu** [comment]
Jenkins test sanity all jdk8,jdk11

**Comment by mpirvu** [comment]
Jenkins test sanity zlinux jdk11

**Comment by mpirvu** [comment]
Tests have passed, hence merging.

**Pull request merged**